frontend/src/pages/EditProfile.jsx

```diff
@@ -1,19 +1,51 @@
 import React, { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
 import './css/EditProfile.css';
 
 const EditProfile = () => {
   const [user, setUser] = useState({
-    username: 'John Doe',
-    email: 'john.doe@example.com',
-    address: '1234 Main Street, City',
-    contact: '+1234567890',
-    bio: 'A passionate college student excited to learn and grow.',
-    country: 'India',
-    academicYear: '2nd Year',
-    profilePicture: 'profile-picture-url',
+    username: '',
+    email: '',
+    address: '',
+    contact: '',
+    bio: '',
+    country: '',
+    academicYear: '',
+    profilePicture: '',
   });
 
   const [profilePictureFile, setProfilePictureFile] = useState(null);
+  const navigate = useNavigate();
+
```

Comment:
keep this function "fetchUserProfile" outside useEffect. just make a function call